webapp/src/components/DonateDirectDebitForm.vue

```diff
@@ -184,32 +179,16 @@
           });
           this.mref = response.data.mref;
           this.done = true;
-        } catch (error) {
-          if (error.response) {
-            // status is not 2xx
-            if (error.response.status < 500 && typeof error.response.data === 'object') {
-              // 3xx or 4xx
-              let extracted = false;
-              if ('email' in error.response.data) {
-                this.email_errors = [error.response.data.email[0]];
-                extracted = true;
-              }
-              if ('amount' in error.response.data) {
-                this.amount_errors = [error.response.data.amount[0]];
-                extracted = true;
-              }
-              // TODO extract more errors
-              if (!extracted) {
-                this.errors = [error.response];
-              }
+        } catch (ex) {
+          let errors = digestError(ex);
+          for (const c in errors) {
+            if (c === 'email') {
+              this.email_errors = errors[c];
+            } else if (c === 'amount') {
+              this.amount_errors = errors[c];
             } else {
-              // 5xx
-              this.errors = ['Something went wrong at the server, but we currently do not know why. The support was already notified.'];
+              this.errors.push(...errors[c]);
             }
```

Comment:
This is something for [`switch ... case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch), similarly in other locations in this commit.

Reply:
next time :)